metadata/metadata_test.go

```diff
@@ -338,6 +339,26 @@ func (s) TestAppendToOutgoingContext_FromKVSlice(t *testing.T) {
 	}
 }
 
+func TestStringerMD(t *testing.T) {
+	for _, test := range []struct {
+		md   MD
+		want []string
```

Comment:
why not an exact string match?

Reply:
Because of the fact that ranging over maps is non-deterministic, therefore the string representation of a map is not stable.

As mentioned above, happy to  change it to print it in alphabetical order of the keys if you prefer.

Reply:
This is fine. I didnt mean to have a exact string matcher test here. Nobody should be relying on the output of a stringer method. 

But I think the only test we want is to make sure that md.String() does not cause a `panic`.  No further action required in this comment. Thanks!